File: Blind75/121. Best Time to Buy and Sell Stock/121. Best Time to Buy and Sell Stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int n = prices.Length;
        int curr_min = prices[0];
        int res = 0;
        foreach(int i in prices){
            curr_min = Math.Min(curr_min, i);
            res = Math.Max(res, i - curr_min);
        }

        return res;
    }
}

File: Blind75/121. Best Time to Buy and Sell Stock/121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0;
        int curr_min = prices[0];
        for(int i=1; i<prices.size(); i++){
            curr_min = min(curr_min, prices[i]);
            res = max(res, prices[i] - curr_min);
        }
        return res;
    }
};
